src/transformers/trainer.py

```diff
@@ -812,7 +812,6 @@ def _maybe_log_save_evalute(self, tr_loss, model, trial, epoch):
         if self.control.should_evaluate:
             metrics = self.evaluate()
             self._report_to_hp_search(trial, epoch, metrics)
-            self.control = self.callback_handler.on_evaluate(self.args, self.state, self.control, metrics)
```

Comment:
Moving this to the end of `evaluate` othewise that even is not called when we call `trainer.evaluate()` independently.